crates/oxc_linter/src/rules/typescript/only_throw_error.rs

```diff
@@ -1,9 +1,34 @@
 use oxc_macros::declare_oxc_lint;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
 
-use crate::rule::Rule;
+use crate::{
+    rule::{DefaultRuleConfig, Rule},
+    utils::{TypeOrValueSpecifier, default_true},
+};
 
 #[derive(Debug, Default, Clone)]
-pub struct OnlyThrowError;
+pub struct OnlyThrowError(Box<OnlyThrowErrorConfig>);
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, JsonSchema)]
+#[serde(rename_all = "camelCase", default)]
+pub struct OnlyThrowErrorConfig {
+    /// An array of type or value specifiers for additional types that are allowed to be thrown.
+    /// Use this to allow throwing custom error types.
+    pub allow: Vec<TypeOrValueSpecifier>,
+    /// Whether to allow throwing values typed as `any`.
+    #[serde(default = "default_true")]
+    pub allow_throwing_any: bool,
+    /// Whether to allow throwing values typed as `unknown`.
+    #[serde(default = "default_true")]
```

Comment:
I can't remember, are these default_trues necessary? I thought the docs were able to determine this via the Default impl automatically.

Reply:
It seems like the docs can generate just fine without it. Although I think it's probably good to include the information on the struct itself. In this case, `Serialize` requires `Default` anyway, so 🤷